### ISSUE TITLE
question on configs

### ISSUE THREAD

**Issue by vedantnilabh** [shenoynikhil/ETFlow#26]
Hi, 
I am trying to replicate your results and was wondering if the configs in the GitHub are the ones you used, or is the schedule in the paper the one you used for training? the max and min learning rates appear to be quite different then the paper (5e-4 and 1e-8 vs 1e-3 and 1e-7) and I dont think the learning rate would decay fully given the number of cycle steps in the config is 375,000 (243573 / 512 = 476 steps per epoch, 476 * 250 =119,000 steps in training). note 512 is the effective batch size (64 * 8). just want to make sure I reproduce fairly

**Comment by shenoynikhil** [comment]
Hey @vedantnilabh,
So the number of steps per epoch is set to 5000 (see `limit_train_batches` in the `trainer` part of the config). I just double checked my wandb run for the drugs dataset case and the max_lr is 5.e-4 and min_lr is 1.e-8.

Also, I recently changes the data processing (see #25, #23) to store data in a slightly different way, so these hyper-parameters might have to change to accurately reproduce. I haven't had the time to exactly reproduce my own results since I have changed the data storage.

**Comment by vedantnilabh** [comment]
hi @shenoynikhil, 
thank you for your quick response. but as a follow up, from looking at the official lightning documentation, limit_train_batches only serves as an upper bound on the number of steps (useful for debugging). Further, when I try to train on your config I also get that behavior (476 steps per epoch), is there something I am missing or doing wrong here? did you additionally modify some behavior to extend your dataset in some way to achieve this, for example, in your data loader?

**Comment by shenoynikhil** [comment]
You're right! Actually my initial dataset was stored a bit in-efficiently as all the conformers of the dataset stored as a single memory map. So the length of the dataset corresponded to total number of conformers (number of unique smiles times 30 (because we use 30 per smiles)). 

I recently made the data loading a bit more efficient (hopefully), so the dataset is now stored in a way that the length corresponds to number of unique molecules (rather than conformers). Therefore, 5000 is quite high. 

For now, you probably will have to train for longer than the current setting specified in the configs. Hope that makes sense.
 
I updated the README to reflect this issue, 
> > [!Important]
> > I have changed some parts of the data preprocessing scripts to make it more efficient. However, these changes might mean that the configs might not lead to the same results as the one reported in the paper. I am working on reproducing the results with the new preprocessed data format. Thanks for your patience.

**Comment by vedantnilabh** [comment]
> You're right! Actually my initial dataset was stored a bit in-efficiently as all the conformers of the dataset stored as a single memory map. So the length of the dataset corresponded to total number of conformers (number of unique smiles times 30 (because we use 30 per smiles)).
> 
> I recently made the data loading a bit more efficient (hopefully), so the dataset is now stored in a way that the length corresponds to number of unique molecules (rather than conformers). Therefore, 5000 is quite high.
> 
> For now, you probably will have to train for longer than the current setting specified in the configs. Hope that makes sense.
> 
> I updated the README to reflect this issue,
> 
> > > [!Important]
> > > I have changed some parts of the data preprocessing scripts to make it more efficient. However, these changes might mean that the configs might not lead to the same results as the one reported in the paper. I am working on reproducing the results with the new preprocessed data format. Thanks for your patience.

Cool, thanks Nikhil! That makes sense!
